pkg/apis/pipeline/v1beta1/param_types.go

```diff
@@ -107,17 +132,23 @@ type ArrayOrString struct {
 	Type      ParamType `json:"type"` // Represents the stored type of ArrayOrString.
 	StringVal string    `json:"stringVal"`
 	// +listType=atomic
-	ArrayVal []string `json:"arrayVal"`
+	ArrayVal  []string          `json:"arrayVal"`
+	ObjectVal map[string]string `json:"objectVal"`
```

Comment:
Now we have 3 options for this struct:
1. Like this current code, we don't rename it
2. Rename it to something else
3. Create a new struct and wrap the ArrayOrString

I think we prefer for 2 right?

Reply:
correct. see the discussion [here](https://github.com/tektoncd/pipeline/pull/4786#discussion_r862023838)